sympy/parsing/latex/lark/transformer.py

```diff
@@ -411,143 +480,269 @@ def inner_product(self, tokens):
         return InnerProduct(Bra(tokens[1]), Ket(tokens[3]))
```

Comment:
I think that every SymPy functions should take `evaluate=False`.
I don't think that we should assume they don't evaluate by default, because the behavior can change in the future.

Reply:
I'll make the change for `Bra`, `Ket`, `InnerProduct`, and `OuterProduct`.

However, while I agree with the sentiment, it is not feasible to do in all cases. In particular, I was passive `evaluate=False` to the `Integral` constructor but it was erroring out because it does not expect that.

Reply:
Why do you have to save the Transformer object to LaTeXParser persistently?

We can construct new ‘Transformer’ with according options just before parsing, than saving it persistently to the parser.

For example, SymPy printer functions already does that, it generates new Printer object every time, and we didn’t get any performance downside of doing that than other performance issues.